### PR TITLE
Enabled custom configuration for attachments

### LINKED ISSUE #608: Ability to specify file size and types for attachments
- In the `Attachments` component there should be some way to specify the file size limit and allowed file types.

cc: @chiraggshah

### LINKED ISSUE #609: Attachments should support single file upload
- The user should only be able to select a single file. (Attachments component)
cc: @chiraggshah

### PR DESCRIPTION
**Pull request by gaagul**
Fixes #608 
Fixes #609 

**Description**

- Added: custom restriction of type and size for files while uploading.
- Added: support to restrict only selection of single file via file input.

**Checklist**

- [x] I have made corresponding changes to the documentation.
- [x] I have added the necessary label (patch/minor/major - If package publish is required).

**Reviewers**

@AbhayVAshokan _a

### CODE REVIEW

**Comment by gaagul** [comment]
@gaagul _a

**Comment by gaagul** [comment]
@AbhayVAshokan _a

File: lib/components/Attachments/index.jsx [hunk 1]
```diff
@@ -151,8 +152,9 @@ const Attachments = (
           />
         )}
         <input
-          multiple
+          accept={config?.allowedFileTypes?.join(",")}
```
Comment (AbhayVAshokan) [inline lib/components/Attachments/index.jsx:155]:
```suggestion
          accept={config.allowedFileTypes?.join(",")}
```
`config` will be non-nullable. It should at least be an empty object.

File: lib/components/Attachments/index.jsx [hunk 2]
```diff
@@ -151,8 +152,9 @@ const Attachments = (
...
           disabled={isUploading}
+          multiple={config?.maxNumberOfFiles !== 1}
```
Comment (AbhayVAshokan) [inline lib/components/Attachments/index.jsx:157]:
```suggestion
          multiple={config.maxNumberOfFiles !== 1}
```

**Review by AbhayVAshokan — CHANGES_REQUESTED** [review]
@gaagul _a https://vimeo.com/797684266/e7bad910f0

**Comment by gaagul** [comment]
> @gaagul _a https://vimeo.com/797684266/e7bad910f0

@AbhayVAshokan There are three ways to implement this,

1. Disable the file input with a condition `disabled={!isEmpty(attachments) && config.maxNumberOfFiles === 1}`
2. Replace the existing file with the new file.
3. Add onClick function to file input
        `onClick={event => {`
            `if (!isEmpty(attachments) && config.maxNumberOfFiles === 1) {`
                `event.preventDefault();`
                `Toastr.error("Only one attachment is allowed");`
            `}`
          `}}`

If the second solution is preferred, It will take more time to implement hence I will have to address this as a separate issue and add it to next week's milestone.

I am moving forward with the third solution as of now. Please share your thoughts on this.

**Comment by gaagul** [comment]
@AbhayVAshokan _a Please review the changes.

**Comment by AbhayVAshokan** [comment]
@gaagul this should be generalized for any number of files. So it would be best if uppy will show an error if we try to upload more then the specified number of files.

https://uppy.io/docs/uppy/#restrictions

**Comment by gaagul** [comment]
> @gaagul this should be generalized for any number of files. So it would be best if uppy will show an error if we try to upload more then the specified number of files.

@AbhayVAshokan I will raise a new issue for this and take it up next week, the problem with any number of files is that file input only supports either one single file or multiple files, We cannot specify the number of files in file input. So I will have to find a way to make changes to the upload functions if we want to allow users to specify an exact number of attachments.

> https://uppy.io/docs/uppy/#restrictions

I think this is mainly used when we are using uppy dashboard. I will look into this feature as a separate issue.

If the current changes are fine, can you please review the changes.

**Review by AbhayVAshokan — APPROVED** [review]
LGTM

**Comment by AbhayVAshokan** [comment]
@gaagul let's pick up the generalization as a new issue. Please create a new issue for the same.

**Pull request merged**